src/features/reports.ts

```diff
@@ -134,7 +142,13 @@ export async function generateReports(): Promise<void> {
       core.summary.write()
     }
     if (arePRReportsEnabled()) {
-      createPRComment(report)
+      if (arePRMergeReportsEnabled()) {
+        const commentId = await findExistingPrCommentId(PR_COMMENT_TITLE)
+        if (commentId) {
+          return updatePRComment(report, commentId)
+        }
+      }
+      return createPRComment(report)
     }
```

Comment:
we could add an else branch that throws an error when the new option is on, because the option "Requires `native-image-pr-reports` to be `true`".